recipe/patches/Modify-install_name_tool-to-work-on-a-file-copy-on-Apple-Silicon.patch

```diff
@@ -0,0 +1,101 @@
+From 552649c0e76178dc070ffe09834953607fc9739a Mon Sep 17 00:00:00 2001
+From: Eli Rykoff <erykoff@stanford.edu>
+Date: Wed, 16 Dec 2020 09:06:12 -0800
+Subject: [PATCH 1/2] Modify install_name_tool to work on a file copy on Apple
+ Silicon
+
+---
+ cctools/misc/install_name_tool.c | 23 ++++++++++++++++++++++-
+ 1 file changed, 22 insertions(+), 1 deletion(-)
+
+diff --git a/cctools/misc/install_name_tool.c b/cctools/misc/install_name_tool.c
+index 080a178..a4b4368 100644
+--- a/cctools/misc/install_name_tool.c
++++ b/cctools/misc/install_name_tool.c
+@@ -26,6 +26,9 @@
+ #include <limits.h>
+ #include <unistd.h>
+ #include <fcntl.h>
++#if defined(__APPLE__) && __aarch64__
++#include <copyfile.h>
++#endif
+ #include "stuff/port.h" /* cctools-port: fake signing */
+ #include "stuff/errors.h"
+ #include "stuff/breakout.h"
+@@ -388,9 +391,22 @@ char *input)
+     struct load_command *lc;
+     enum byte_sex host_byte_sex;
+ 
+-	host_byte_sex = get_host_byte_sex();
++#if defined(__APPLE__) && __aarch64__
++        char tempfile[PATH_MAX];
++
++	snprintf(tempfile, PATH_MAX, "%sXXXXXX", input);
++	mktemp(tempfile);
+ 
++	if (copyfile(input, tempfile, NULL, COPYFILE_ALL | COPYFILE_MOVE) == -1)
++	  system_error("failed to copy %s for installation", input);
++
++	fd = open(tempfile, O_WRONLY, 0);
++#else
+ 	fd = open(input, O_WRONLY, 0);
++#endif
++
++	host_byte_sex = get_host_byte_sex();
++
+ 	if(fd == -1)
+ 	    system_error("can't open input file: %s for writing", input);
+ 
+@@ -454,6 +470,11 @@ char *input)
+ 	}
+ 	if(close(fd) == -1)
+ 	    system_error("can't close written on input file: %s", input);
++
++#if defined(__APPLE__) && __aarch64__
++	if (rename(tempfile, input) == -1)
++	  system_error("can't rename tempfile to %s", input);
++#endif
+ }
+ 
+ /*
+
+From da8726371d48409f4235dcd7c7c9dd30b79c9e6f Mon Sep 17 00:00:00 2001
+From: Eli Rykoff <erykoff@stanford.edu>
+Date: Thu, 17 Dec 2020 21:33:02 -0800
+Subject: [PATCH 2/2] Add /* cctools-port */ comments
+
+---
+ cctools/misc/install_name_tool.c | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/cctools/misc/install_name_tool.c b/cctools/misc/install_name_tool.c
+index a4b4368..cc80ce2 100644
+--- a/cctools/misc/install_name_tool.c
++++ b/cctools/misc/install_name_tool.c
+@@ -26,7 +26,7 @@
+ #include <limits.h>
+ #include <unistd.h>
+ #include <fcntl.h>
+-#if defined(__APPLE__) && __aarch64__
++#if defined(__APPLE__) && __aarch64__ /* cctools-port: copy before modification */
+ #include <copyfile.h>
+ #endif
+ #include "stuff/port.h" /* cctools-port: fake signing */
+@@ -391,7 +391,7 @@ char *input)
+     struct load_command *lc;
+     enum byte_sex host_byte_sex;
+ 
+-#if defined(__APPLE__) && __aarch64__
++#if defined(__APPLE__) && __aarch64__ /* cctools-port: copy before modification */
+         char tempfile[PATH_MAX];
+ 
+ 	snprintf(tempfile, PATH_MAX, "%sXXXXXX", input);
```

Comment:
I guess you used unistd.h for this prototype.  Actually the proper include here is #include <stdio.h> instead for having this prototype on unix. For windows-sake you should addtionally include stdlib.h ... but in general stdio.h should be fine here too.